pkg/sql/pg_catalog_test.go

```diff
@@ -0,0 +1,379 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+// This test compares a dump generated from PostgreSQL with the pg_catalog
+// and compares it with the current pg_catalog at cockroach db skipping
+// all the known diffs:
+//
+// cd pkg/sql
+// go test -run TestPGCatalog
+//
+// If you want to re-create the known (expected) diffs with the current differences
+// add -rewrite-diffs flag when running this test:
+//
+// cd pkg/sql
+// go test -run TestPGCatalog -rewrite-diffs
+//
+// To create the postgres dump file see pkg/cmd/generate-pg-catalog/main.go:
+//
+// cd pkg/cmd/generate-pg-catalog/
+// go run main.go > ../../sql/testdata/pg_catalog_tables
+package sql
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/errors/oserror"
+)
+
+// This constants enumerates indexes in pg_catalog_tables
+const (
+	testTableIndex = iota
+	testColumnIndex
+	testTypeIndex
+	testTypeOidIndex
+)
+
+// This query uses udt_name::regtype instead of data_type column because
+// data_type only says "ARRAY" but does not say which kind of array it is.
+const getPgCatalogSQL = `
+	SELECT table_name, column_name, udt_name::regtype data_type, udt_name::regtype::Oid data_type_oid
```

Comment:
this query should also be updated like in my comment from the other file

Reply:
Done